Create multiple-config-instances.md (#900)
ZupIT/beagle-docs: content/en/android/tutorials/multiple-config-instances.md

---
title: How to use multiple instances of beagle
weight: 234
description: >-
  On this section, you'll find the step by step to start beagle configuration for multiple instances
---

---
### What is it? 

> The **Beagle Multiple Instances** is a way to start beagle for the current server driven flow using another beagle configuration, such as new widgets, actions and so on.

Reviewer: ```suggestion
**Beagle Multiple Instances** is a feature that allows the start up of distinct server driven flows with independent configurations of Beagle. This allows each server driven flow to have its own components, base url, actions, http client, etc.
```

> You will see below how to configure the multiple instances.

Reviewer: ```suggestion

```

        it.logInfo("$CONFIG_NAME:analyticsProvider")
        AppAnalytics()
    }

    override val urlBuilder = beagleConfigFactory<UrlBuilder> {
        it.logInfo("$CONFIG_NAME:urlBuilder")
        AppUrlBuilder()
    }

    override fun registeredWidgets() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredWidgets")
            listOf(
                Text3::class.java as Class<WidgetView>,
                Input::class.java as Class<WidgetView>)
        }

    override fun registeredActions() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredActions")
            listOf(
                br.com.zup.beagle.sample.actions.CustomAndroidAction::class.java as Class<Action>,
            )
        }

    override fun registeredOperations() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredOperations")
            mapOf<String, Operation>(
                "isValidCpf" to br.com.zup.beagle.sample.operations.IsValidCPFOperation(),

                )
        }
}
```

### Configuring a beagle entry point to use the new configuration

> A function called [**newServerDrivenIntent**]({{< ref path="/android/tutorials/how-to-display-a-screen" lang="en" >}}) is responsible to display a server-driven screen, now you can define that this screen will use the configuration defined above like this:

Reviewer: ```suggestion
The function [**newServerDrivenIntent**]({{< ref path="/android/tutorials/how-to-display-a-screen" lang="en" >}}) is responsible for displaying a server-driven screen. To define that this screen must use the configuration defined above, do:
```
